src/algebra/order/field.lean

```diff
@@ -725,4 +726,34 @@ lemma one_div_pow_mono (a1 : 1 ≤ a) : monotone (λ n : ℕ, order_dual.to_dual
 lemma one_div_pow_strict_mono (a1 : 1 < a) : strict_mono (λ n : ℕ, order_dual.to_dual 1 / a ^ n) :=
 λ m n, one_div_pow_lt_one_div_pow_of_lt a1
 
+/-! ### Results about `has_lub` and `has_glb` -/
+
+lemma is_lub_mul {s : set α} (hc : 0 ≤ a) (hs : is_lub s b) :
+  is_lub {x | ∃ y ∈ s, a * y = x} (a * b) :=
+begin
+  rcases lt_or_eq_of_le hc with hc | rfl,
+  { rw ← (order_iso.mul_left₀ _ hc).is_lub_image' at hs,
+    convert hs using 1,
+    ext x,
+    simp },
+  { convert is_lub_singleton using 1,
+    ext x,
+    have : s.nonempty ∧ 0 = x ↔ x = 0 := by rw [and_iff_right hs.nonempty, eq_comm],
+    simpa using this },
+end
+
+lemma is_glb_mul {s : set α} (hc : 0 ≤ a) (hs : is_glb s b) :
+  is_glb {x | ∃ y ∈ s, a * y = x} (a * b) :=
+begin
+  rcases lt_or_eq_of_le hc with hc | rfl,
+  { rw ← (order_iso.mul_left₀ _ hc).is_glb_image' at hs,
+    convert hs using 1,
+    ext x,
+    simp },
+  { convert is_glb_singleton using 1,
+    ext y,
+    have : s.nonempty ∧ 0 = y ↔ y = 0 := by rw [and_iff_right hs.nonempty, eq_comm],
+    simpa using this },
+end
```

Comment:
I think it would be good to have these lemmas:
```lean
@[simp] lemma _root_.set.image_const' {α β : Sort*} (s : set α) (b : β) (hs : s.nonempty) :
  (λ a, b) '' s = {b} :=
begin
  ext,
  rw [set.mem_image, set.mem_singleton_iff, eq_comm],
  simp only [and_iff_right_iff_imp, exists_and_distrib_right],
  intro h,
  exact hs,
end

@[simp] lemma _root_.set.image_const {α β : Sort*} (s : set α) (b : β) (hs : s.nonempty) :
  (function.const _ b) '' s = {b} :=
set.image_const' s b hs
```
which I think would reduce the second branch here.

Reply:
Nevermind, they exist already!